src/bun.js/builtins/js/JSBufferPrototype.js

```diff
@@ -297,7 +297,14 @@ function slice(start, end) {
       end = end || this.byteLength;
   }
 
-  return new Buffer(this.buffer, this.byteOffset + start, end - start);
+  let len = end - start;
+  start = this.byteOffset + start;
+
+  if (len > this.byteLength) {
+    len = this.byteLength - start;
+  }
+
+  return new Buffer(this.buffer, start, len);
```

Comment:
i think this logic should be housed in the Buffer ctor since subarray and copy methods also use it.